### PR TITLE
Apache Flink Connector for OpenSearch

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Apache Flink Connector for OpenSearch blog post
 
### Issues Resolved
Closes https://github.com/opensearch-project/project-website/issues/1233

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by reta** [comment]
@brijos @ftisiot @kolchfa-aws fyi

**Comment by brijos** [comment]
Do we have @ftisiot's author bio and image?

**Comment by ftisiot** [comment]
Yep, it's here! https://github.com/opensearch-project/project-website/pull/1280

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+* Submit a pull request to https://github.com/apache/flink-connector-opensearch/
+* Optionally, once pull request is merged, ask for connector release by [following release process procedure](https://cwiki.apache.org/confluence/display/FLINK/Creating+a+flink-connector+release)
+
+Apache Flink has a large community and is being actively developed. It might happen that reviewing and merging changes could take some time but that should not discourage anyone from contributing. One of the interesting new features being developed is the implementation of a new generation of sinks, based on [FLIP-171: Async Sink](https://cwiki.apache.org/confluence/display/FLINK/FLIP-171%3A+Async+Sink) proposal. The [related pull request](https://github.com/apache/flink-connector-opensearch/pull/5) is already open, please feel free to check it out and contribute! 
+
+If you want to get involved or just stay aware of what is happening in Apache Flink community, please consider subscribing to the [mailing list](https://flink.apache.org/community.html#how-to-subscribe-to-a-mailing-list) or joining the official [Slack channel](https://flink.apache.org/community.html#slack).
```
Comment (kolchfa-aws) [inline _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:190]:
```suggestion
If you want to get involved or just be aware of what is happening in the Apache Flink community, please consider subscribing to the [mailing list](https://flink.apache.org/community.html#how-to-subscribe-to-a-mailing-list) or joining the official [Slack channel](https://flink.apache.org/community.html#slack). 
```

**Review by kolchfa-aws — COMMENTED** [review]
Thank you for writing this blog! This is wonderful. I left some suggestions; please review and let me know if you'd like to discuss. After that we'll just need an editorial review.

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+```
+docker run -d                           \
+    -p 9200:9200                        \
+    -p 9600:9600                        \
+    -e "discovery.type=single-node"     \
+    opensearchproject/opensearch:2.4.1
+```
+
+The latest OpenSearch version as of today is **2.4.1**, however the [Apache Flink Connector for OpenSearch](https://github.com/apache/flink-connector-opensearch/) can also be used with the older **1.x** release line.
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:83]: Seems like 2.5 is a bit behind, still no official announcement, I will add the suggestion and update OpenSearch with 2.5.0 once it is out.

[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown]: Updated, thanks @kolchfa-aws !

[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown]: 2.5.0 tag was just pushed (https://github.com/opensearch-project/OpenSearch/releases/tag/2.5.0), updated the release

**Review by natebower — COMMENTED** [review]
Hi @reta, I'm the editor for the OpenSearch Project. Please see my comments and changes, and feel free to let me know if you have any questions or would like to discuss. I'm happy to clarify any of my suggested changes for you, if you need. Once you've had a chance to address my changes/comments, please tag me so that I can give this a final skim and move it forward in our process. Thanks!

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+```
+docker run -d                           \
+    -p 9200:9200                        \
+    -p 9600:9600                        \
+    -e "discovery.type=single-node"     \
+    opensearchproject/opensearch:2.5.0
+```
+
+The latest OpenSearch version as of today is **2.5.0**, however you can use [Apache Flink Connector for OpenSearch](https://github.com/apache/flink-connector-opensearch/) with any `1.x` or `2.x` OpenSearch version.
```
[the remark this answers is not in the document]
Comment (pajuric) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:83]: @kolchfa-aws - Please link to the OpenSearch 2.5.0 blog: https://opensearch.org/blog/opensearch-2-5-is-live/

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+The process is very simple:
+
+* Create a [JIRA issue](https://issues.apache.org/jira/projects/FLINK)
+* Submit a pull request to the [Apache Flink Connector for OpenSearch repo](https://github.com/apache/flink-connector-opensearch/)
+* Optionally, once pull request is merged, ask for a connector release by following the [release process](https://cwiki.apache.org/confluence/display/FLINK/Creating+a+flink-connector+release)
+
+Apache Flink has a large community and is being actively developed. Even though reviewing and merging changes may take some time, it should not discourage anyone from contributing. One of the interesting new features being developed is the implementation of a new generation of sinks based on [FLIP-171: Async Sink](https://cwiki.apache.org/confluence/display/FLINK/FLIP-171%3A+Async+Sink) proposal. The [related pull request](https://github.com/apache/flink-connector-opensearch/pull/5) is already open, so please feel free to check it out and contribute!
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:201]: @natebower the suggestion is changing the meaning of the sentence. "Apache Flink has a large community and is being actively developed" - the emphasis here is that project is not dead and under active development. Changing that to "Flink has a large and actively developing community" reads like community is being developed?

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+layout: post
+title:  "Apache Flink® connector to OpenSearch®"
+authors:
+- ftisiot
+- reta
+date: 2022-01-31
+categories:
+ - technical-post
+
+excerpt: "Apache Flink® and OpenSearch® are widely known and successful open source projects. Even if you have never used them, it is very likely you have heard or read about them in the search or streaming data transformation use-cases. And there are reasons for that: although they are targeting different tech markets, they perform their job very well."
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:11]: I mean "use cases" are general streaming data transformations: take from one place, transform, put into another place. I am not sure we could clarify without going into concrete problem.

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+![Flink + OpenSearch](flink_opensearch.png)
+
+## What is Flink?
+
+From the [official website](https://flink.apache.org/) of the project:
+
+> Apache Flink is a framework and distributed processing engine for stateful computations over unbounded and bounded data streams. Flink has been designed to run in all common cluster environments, perform computations at in-memory speed and at any scale.
+
+Apache Flink started from a fork of Stratosphere's distributed execution engine and became an Apache Incubator project in March 2014. In December 2014, Flink was accepted as an Apache top-level project.
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:26]: I believe we need full project name to not confuse readers

**Comment by reta** [comment]
> Hi @reta, I'm the editor for the OpenSearch Project.

Thanks a lot for the review, @natebower ! Applied the majority of them, however have concerns regarding Apache Flink -> Flink changes, I believe it is beneficial to use full project name all the time, @ftisiot what do you think?

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 4 above)
[the remark this answers is not in the document]
Comment (natebower) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:201]: @reta That makes sense. In that case, how about "Flink has a large community, and the project is being actively developed"?

**Comment by ftisiot** [comment]
> > Hi @reta, I'm the editor for the OpenSearch Project.
> 
> Thanks a lot for the review, @natebower ! Applied the majority of them, however have concerns regarding Apache Flink -> Flink changes, I believe it is beneficial to use full project name all the time, @ftisiot what do you think?

I would keep Apache Flink all the time if that's not a problem

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 5 above)
[the remark this answers is not in the document]
Comment (natebower) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:11]: @reta Ok, if we're just referring to use cases generally, then we should probably remove "the", which implies that we're referring to some specific use cases. Would "...it is likely you have heard or read about them in search or streaming data transformation use cases" work?

**Comment by natebower** [comment]
> > Hi @reta, I'm the editor for the OpenSearch Project.
> 
> Thanks a lot for the review, @natebower ! Applied the majority of them, however have concerns regarding Apache Flink -> Flink changes, I believe it is beneficial to use full project name all the time, @ftisiot what do you think?

@reta I was going with [this page](https://flink.apache.org/flink-architecture.html), where they seem to largely use just "Flink" (though they are not completely consistent), but I'll defer to you and @ftisiot on this.

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 5 above)
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:11]: Apologies, that's what the current wording says:

> Even if you have never used them, it is very likely you have heard or read about them in the search or streaming data transformation use-cases.

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+To start defining the first data pipelines, it might be helpful to use the highest level of abstraction in Flink represented by its SQL layer. You can play with a [Docker version of Flink](https://github.com/aiven/sql-cli-for-apache-flink-docker), which offers bare-bones capabilities, or explore a slicker experience on [Aiven for Apache Flink](https://aiven.io/flink?utm_source=blog&utm_medium=organic&utm_campaign=blog_opensearch_flink_opensearch_connector).
+
+If you’re looking for a practical example, check out [how to build a real-time alerting solution](https://docs.aiven.io/docs/products/flink/howto/real-time-alerting-solution) with Apache Flink and a few SQL statements.
+
+## What does the Apache Flink Connector for OpenSearch do?
+
+The [Apache Flink Connector for OpenSearch](https://github.com/apache/flink-connector-opensearch/) allows writing from Apache Flink into an index of the [OpenSearch](https://opensearch.org/) engine (sink side). It does not support reading from the index (source side). The connector is quite a recent addition to the long list of connectors supported by Apache Flink and is available starting from release **1.16**.
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:67]: I don't think so, example from similar domains: "MySQL table", "MongoDB collection" - the engine is certainly redundant

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+  'allow-insecure' = 'true',
+  'index' = 'users'
+);
+```
+
+The `'connector' = 'opensearch'` parameter defines the type of connector. The `hosts`, `username` and `password` define the target OpenSearch endpoint and authentication credentials.
+By default, OpenSearch distributions come with security turned on, meaning communication over HTTPS and mandatory username / password authentication. The `allow-insecure` connector option allows connecting to clusters which use self-signed certificates. 
+The `index` parameter defines the target OpenSearch index.
+
+What happens if we insert some data into the table?
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:110]: > is there a way we can rephrase it to match the tone of the rest of the example?

How about that: "You may be wondering what happens if we insert some data into the table? Let us see."

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 6 above)
[the remark this answers is not in the document]
Comment (ftisiot) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:26]: I would, for consistency, always refer to Apache Flink

[the remark this answers is not in the document]
Comment (ftisiot) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:26]: but this is a personal preference, not a must have

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+
+> Apache Flink is a framework and distributed processing engine for stateful computations over unbounded and bounded data streams. Flink has been designed to run in all common cluster environments, perform computations at in-memory speed and at any scale.
+
+Apache Flink started from a fork of Stratosphere's distributed execution engine and became an Apache Incubator project in March 2014. In December 2014, Flink was accepted as an Apache top-level project.
+
+Flink is both a **framework**, providing the basics to create apps in Java, Python, and Scala, and a **distributed processing engine**, which can scale vertically and horizontally.
+It is used to create data pipelines over both unbounded (e.g. streaming) and bounded (batch) datasets, offering various levels of abstractions depending on the computation needs.
+Data is sourced and sinked via dedicated pluggable connectors, similarly to Apache Kafka. The Flink connector ecosystem is continuously evolving with new technologies being added regularly.
+
+All the principles listed above make Apache Flink a go-to open source data processing solution for a wide variety of industries and use cases.
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:32]: I would argue those are "principles" (or design choices if you will) but not "features".

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 8 above)
[the remark this answers is not in the document]
Comment (natebower) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:110]: I like it! But no question mark, and let's use a contraction at the end: "You may be wondering what happens if we insert some data into the table. Let's see."

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 9 above)
[the remark this answers is not in the document]
Comment (natebower) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:32]: Ok, let's go with "All of these principles". We want to avoid "above" for accessibility purposes.

**Comment by reta** [comment]
> I would keep Apache Flink all the time if that's not a problem

@ftisiot @natebower  Updated to use Apache Flink everywhere

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 7 above)
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:67]: Got it, changed to "OpenSearch index"

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+
+## Why did Aiven choose Flink over other options?
+
+Aiven applies dogfooding: Aiven runs on Aiven, meaning that we operate on the same data infrastructure we offer to our clients. Our standard approach is to evaluate open-source solutions for our internal needs, gain experience, make them robust and then, once we’re confident, offer them to our clients. 
+
+The same happened with Apache Flink, which was chosen because it’s open source (Aiven loves open source and we have a [dedicated OSPO team](https://aiven.io/blog/aivens-open-source-program-office)), and because it has a great and variegated community.
```
[the remark this answers is not in the document]
Comment (ftisiot) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:38]: What about?

> We took the same approach with Flink, which was chosen because it’s open source (Aiven loves open source, and we have a [dedicated OSPO team](https://aiven.io/blog/aivens-open-source-program-office)) and because it has a large, varied and supportive community.

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+* Unifies unbounded and bounded data pipelines, making the transition from batch to streaming a matter of just redefining data sources/sinks
+* Decouples compute from data storage, meaning that it's possible to change the backend without needing to re-architect the transformation layer
+* Scales both vertically and horizontally to accommodate large workloads
+* Offers a rich SQL interface that covers all our data transformations needs
+* Offers a variety of language SDKs at various abstraction levels
+* Works natively with other open-source tools like PostgreSQL®, Apache Kafka® and now OpenSearch
+
+## How does Aiven use Flink?
+
+The spread of [Apache Flink’s SQL functions](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/functions/systemfunctions/) means that Flink can be used to define a vast variety of data pipelines. Apart from the traditional analytical or filtering workloads, Aiven is using or planning to use Flink for two main use cases:
```
[the remark this answers is not in the document]
Comment (ftisiot) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:51]: Fine by me

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 10 above)
[the remark this answers is not in the document]
Comment (natebower) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:38]: @ftisiot Sounds great! Let's add the serial comma after "varied", though.

[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:38]: @ftisiot could you please make it a suggestion so I would make the right change, thank you.

[the remark this answers is not in the document]
Comment (ftisiot) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:38]:
```suggestion
We took the same approach with Flink, which was chosen because it’s open source (Aiven loves open source, and we have a [dedicated OSPO team](https://aiven.io/blog/aivens-open-source-program-office)) and because it has a large, varied, and supportive community.
```

**Comment by reta** [comment]
@natebower I think I went though all of the comments :bow: , please let me know if I missed something (not on purpose) :D

**Review by natebower — DISMISSED** [review]
@reta LGTM, thanks for the collaboration! @pajuric Approved and ready to publish.

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+authors:
+- ftisiot
+- reta
+date: 2022-01-31
+categories:
+ - technical-post
+
+excerpt: "Apache Flink® and OpenSearch® are widely known and successful open source projects. Even if you have never used them, it is likely you have heard or read about them in search or streaming data transformation use cases. And there are reasons for that: although they target different tech markets, they perform their job very well."
+---
+
```
Comment (pajuric) [inline _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:13]: Please add front matter here using the provided below: 

**Meta Keywords:** Apache Flink Connector to OpenSearch, open source data processing, Aiven for Apache Flink
**Meta Description:**  Learn how Aiven uses the Apache Flink Connector data processing solution to write from Apache Flink to an index of the OpenSearch engine.

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+To start defining the first data pipelines, it might be helpful to use the highest level of abstraction in Flink, represented by its SQL layer. You can experiment with a [Docker version of Flink](https://github.com/aiven/sql-cli-for-apache-flink-docker), which offers bare-bones capabilities, or explore a slicker experience on [Aiven for Apache Flink](https://aiven.io/flink?utm_source=blog&utm_medium=organic&utm_campaign=blog_opensearch_flink_opensearch_connector).
+
+If you’re looking for a practical example, check out [how to build a real-time alerting solution](https://docs.aiven.io/docs/products/flink/howto/real-time-alerting-solution) with Apache Flink and a few SQL statements.
+
+## What does the Apache Flink Connector for OpenSearch do?
+
+The [Apache Flink Connector for OpenSearch](https://github.com/apache/flink-connector-opensearch/) allows writing from Apache Flink into [OpenSearch](https://opensearch.org/) index (sink side). It does not support reading from the index (source side). The connector is a recent addition to the long list of connectors supported by Apache Flink and is available starting with release **1.16**.
```
Comment (natebower) [inline _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:69]:
```suggestion
The [Apache Flink Connector for OpenSearch](https://github.com/apache/flink-connector-opensearch/) allows writing from Apache Flink into an [OpenSearch](https://opensearch.org/) index (sink side). It does not support reading from the index (source side). The connector is a recent addition to the long list of connectors supported by Apache Flink and is available starting with release **1.16**. 
```

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+title:  "Apache Flink® Connector for OpenSearch®"
+authors:
+- ftisiot
+- reta
+date: 2022-01-31
+categories:
+ - technical-post
+
+excerpt: "Apache Flink® and OpenSearch® are widely known and successful open source projects. Even if you have never used them, it is likely you have heard or read about them in search or streaming data transformation use cases. And there are reasons for that: although they target different tech markets, they perform their job very well."
+meta_keywords: "Apache Flink Connector to OpenSearch, open source data processing, Aiven for Apache Flink"
```
Comment (natebower) [inline _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:12]:
```suggestion
meta_keywords: "Apache Flink Connector for OpenSearch, open-source data processing, Aiven for Apache Flink"
```

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+meta_description: "Learn how Aiven uses the Apache Flink Connector data processing solution to write from Apache Flink to an index of the OpenSearch engine."
```
Comment (natebower) [inline _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:13]:
```suggestion
meta_description: "Learn how Aiven uses the Apache Flink Connector data processing solution to write from Apache Flink into an OpenSearch index."
```

**Comment by natebower** [comment]
Just a few more minor commits, please and thanks!

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 16]
```diff
@@ -0,0 +1,208 @@
...
+
+```
+docker run -d                           \
+    -p 9200:9200                        \
+    -p 9600:9600                        \
+    -e "discovery.type=single-node"     \
+    opensearchproject/opensearch:2.5.0
+```
+
+The latest OpenSearch version is **2.5.0**; however, you can use the [Apache Flink Connector for OpenSearch](https://github.com/apache/flink-connector-opensearch/) with any `1.x` or `2.x` OpenSearch version.
```
Comment (kolchfa-aws) [inline _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:85]: @reta: could you please update this sentence with a link to the blog:
```suggestion
The latest OpenSearch version is [**2.5.0**](https://opensearch.org/blog/opensearch-2-5-is-live/); however, you can use the [Apache Flink Connector for OpenSearch](https://github.com/apache/flink-connector-opensearch/) with any `1.x` or `2.x` OpenSearch version. 
```

**Review by jhmcintyre — CHANGES_REQUESTED** [review]

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 17]
```diff
@@ -0,0 +1,208 @@
+---
+layout: post
+title:  "Apache Flink® Connector for OpenSearch®"
```
[the remark this answers is not in the document]
Comment (jhmcintyre) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:3]: @reta for visibility. Please let me know of any questions.

[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:3]: @jhmcintyre Following the ASF trademark guide we use to add the ® to the most prominent and first text occurrence of a product [1]

[1] https://www.apache.org/foundation/marks/#principles

[the remark this answers is not in the document]
Comment (jhmcintyre) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:3]: Thanks @reta. I read the Apache guidelines as referring to using the full name (i.e. rather than just "Flink") on first and most prominent usage, rather that specifying inclusion of trademark symbols, per [2]. That said, for this post, can we agree on including trademark symbols on the first usage in the body copy, and removing from headline and later appearances in body copy that I've flagged?

[2] https://www.apache.org/foundation/marks/guide

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 18]
```diff
@@ -0,0 +1,208 @@
...
+authors:
+- ftisiot
+- reta
+date: 2022-01-31
+categories:
+ - technical-post
+
+excerpt: "Apache Flink® and OpenSearch® are widely known and successful open source projects. Even if you have never used them, it is likely you have heard or read about them in search or streaming data transformation use cases. And there are reasons for that: although they target different tech markets, they perform their job very well."
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:11]: :+1:

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown (hunk 17 above)
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:3]: :+1:

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 19]
```diff
@@ -0,0 +1,208 @@
...
+meta_keywords: "Apache Flink Connector for OpenSearch, open-source data processing, Aiven for Apache Flink"
+meta_description: "Learn how Aiven uses the Apache Flink Connector data processing solution to write from Apache Flink into an OpenSearch index."
+---
+
+Apache Flink® and OpenSearch® are widely known and successful open source projects. Even if you have never used them, it is likely you have heard or read about them in search or streaming data transformation use cases. And there are reasons for that: although they target different tech markets, they perform their job very well.
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:16]: This is a first usage, keeping it as per https://github.com/opensearch-project/project-website/pull/1300#discussion_r1096446340

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 20]
```diff
@@ -0,0 +1,208 @@
...
+We took the same approach with Flink, which was chosen because it’s open source (Aiven loves open source, and we have a [dedicated OSPO team](https://aiven.io/blog/aivens-open-source-program-office)) and because it has a large, varied, and supportive community.
+
+On the technical side, Flink has all the qualifications to become the de facto standard in data processing because it:
+
+* Unifies unbounded and bounded data pipelines, making the transition from batch to streaming a matter of just redefining data sources/sinks.
+* Decouples compute from data storage, meaning that it's possible to change the backend without needing to re-architect the transformation layer.
+* Scales both vertically and horizontally to accommodate large workloads.
+* Offers a rich SQL interface that covers all our data transformations needs.
+* Offers a variety of language SDKs at various abstraction levels.
+* Works natively with other open-source tools like PostgreSQL®, Apache Kafka®, and now OpenSearch.
```
[the remark this answers is not in the document]
Comment (reta) [reply _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:49]: As per https://github.com/opensearch-project/project-website/pull/1300#discussion_r1096446340, keeping trademarks on the first usage

**Comment by krisfreedain** [comment]
@reta - can you fix DCO on this? Once we have that resolved, we can push this out.

**Comment by pajuric** [comment]
@reta - Are we good to push this live? If so, please work with @krisfreedain to publish.

**Comment by reta** [comment]
@krisfreedain @pajuric squashed and fixed DCO, thank you!

**Comment by reta** [comment]
Yes, we are good to go!

File: _posts/2023-01-23-OpenSearch-Flink-Connector.markdown [hunk 21]
```diff
@@ -0,0 +1,208 @@
+---
+layout: post
+title:  "Apache Flink Connector for OpenSearch"
+authors:
+- ftisiot
+- reta
+date: 2022-01-31
```
Comment (krisfreedain) [inline _posts/2023-01-23-OpenSearch-Flink-Connector.markdown:7]: sorry - one more edit - can you edit the date? @reta

**Review by krisfreedain — CHANGES_REQUESTED** [review]
one small date change

**Comment by pajuric** [comment]
@reta one more quick tweak and we should be good to go

**Comment by krisfreedain** [comment]
@reta - we're in 2023 not 2022  ;)  can you fix that
